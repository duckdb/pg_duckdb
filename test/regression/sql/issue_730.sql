CREATE TEMP TABLE t_ddb(a int) USING duckdb;
INSERT INTO t_ddb VALUES (1);

CREATE OR REPLACE FUNCTION f2() RETURNS void
    LANGUAGE plpgsql
    RETURNS NULL ON NULL INPUT
    AS
$$
BEGIN
    CREATE TEMP TABLE t_ddb2(a int) USING duckdb;
    CREATE TEMP TABLE t_ddb3(a) USING duckdb AS SELECT 1;
    DROP TABLE t_ddb3;
    INSERT INTO t_ddb2 VALUES (1);
    ALTER TABLE t_ddb2 ADD COLUMN b int;
    ALTER TABLE t_ddb2 ADD COLUMN c int;
    ALTER TABLE t_ddb2 ADD COLUMN d int DEFAULT 100, ADD COLUMN e int DEFAULT 10;
    ALTER TABLE t_ddb2 RENAME COLUMN b TO f;
    ALTER TABLE t_ddb2 RENAME TO t_ddb4;
END;
$$;

SELECT * FROM f2();
SELECT * FROM f2();
