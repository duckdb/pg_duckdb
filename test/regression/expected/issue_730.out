CREATE TEMP TABLE t_ddb(a int) USING duckdb;
INSERT INTO t_ddb VALUES (1);
CREATE OR REPLACE FUNCTION f2() RETURNS void
    LANGUAGE plpgsql
    RETURNS NULL ON NULL INPUT
    AS
$$
BEGIN
    CREATE TEMP TABLE t_ddb2(a int) USING duckdb;
    CREATE TEMP TABLE t_ddb3(a) USING duckdb AS SELECT 1;
    DROP TABLE t_ddb3;
    INSERT INTO t_ddb2 VALUES (1);
    ALTER TABLE t_ddb2 ADD COLUMN b int;
    ALTER TABLE t_ddb2 ADD COLUMN c int;
    ALTER TABLE t_ddb2 ADD COLUMN d int DEFAULT 100, ADD COLUMN e int DEFAULT 10;
    ALTER TABLE t_ddb2 RENAME COLUMN b TO f;
    ALTER TABLE t_ddb2 RENAME TO t_ddb4;
END;
$$;
SELECT * FROM f2();
WARNING:  (PGDuckDB/CreatePlan) Prepared query returned an error: 'Catalog Error: Table Function with name f2 does not exist!
Did you mean "sniff_csv"?

LINE 1: SELECT f2 FROM f2() f2(f2)
                       ^
 f2 
----
 
(1 row)

SELECT * FROM f2();
WARNING:  (PGDuckDB/CreatePlan) Prepared query returned an error: 'Catalog Error: Table Function with name f2 does not exist!
Did you mean "sniff_csv"?

LINE 1: SELECT f2 FROM f2() f2(f2)
                       ^
ERROR:  relation "t_ddb4" already exists
CONTEXT:  SQL statement "ALTER TABLE t_ddb2 RENAME TO t_ddb4"
PL/pgSQL function f2() line 11 at SQL statement
